flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1441,13 +1441,7 @@ private void preCheckBeforeTriggeringCheckpoint(boolean isPeriodic, boolean forc
 		}
 
 		if (!forceCheckpoint) {
-			if (triggerRequestQueued) {
-				LOG.warn("Trying to trigger another checkpoint for job {} while one was queued already.", job);
-				throw new CheckpointException(CheckpointFailureReason.ALREADY_QUEUED);
-			}
-
```

Comment:
And why is this no longer possible?

Reply:
It's possible. But since I have refactored the `triggerRequestQueued` to `periodicTriggeringSuspended`, it makes no more sense to check `periodicTriggeringSuspended`. If the trigger could satisfy other requirements, it could be executed. 

Actually in the current implementation. It's also about the non-periodic checkpoint triggering.
1. Savepoint does not care this checking, because  the `forceCheckpoint` is set to true.
2. Periodic checkpoint trigger would not reach here because it would be canceled if `triggerRequestQueued` is set to true. (There should be no in-flight periodic trigger since it's single-threaded.)
So it's also dead code...